docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+# How-To: Develop an F' Subtopology
+
+Subtopologies are topologies for smaller chunks of behavior in F Prime. It allows for grouping bits of topology architecture that fit together, to then be imported into a base deployment's topology. The use case for this is seen when working with shareable components, specifically in the form of [libraries](./develop-fprime-libraries.md).
+
+*Contents*
+1. [Subtopology Structure](#subtopology-structure)
+2. [Individual File Contents](#individual-file-contents) (this is a long section with multiple subsections!)
+3. [Integration into a "Main" Deployment](#integration-into-a-main-deployment)
+4. [Conclusion](#conclusion)
+
+## Subtopology Structure
+
+A subtopology is a folder that contains a `topology.cpp` file, which includes the behavior for setting up your custom topology. It additionally contains a `topology.fpp` file, which is a *unified* topology file. As opposed to having separate files for component instances and the topology itself, the single file encapsulates both. A `topologydefs.hpp` defines a struct that is used when instantiating your subtopology. A `CMakeLists.txt` file links your `topology.cpp` file to the build for your project.
+
+Thus, the required structure for your subtopology should be:
+
+```bash
+MySubtopology/
+├─ CMakeLists.txt
+├─ MySubtopology.cmake
+├─ MySubtopology.fpp
+├─ MySubtopologyTopology.cpp
+├─ MySubtopologyTopology.hpp
+└─ MySubtopologyTopologyDefs.hpp
+```
+
+All files will be discussed in more detail in later sections of this guide. Additionally, note that there are optional files that can be included in your subtopology to extend its capability.
+
+- It is highly recommended to include a `docs` folder to document your subtopology. Simple markdown files work well in this case.
+- `.fppi` files can be included as config files to your subtopology. They can include useful constants or structures that allow users to modify your subtopology.
```

Comment:
I don't think we document what `.fppi` is or stands for in our docs. Could be good to clarify that `i` is just a naming convention for include, or simply say additional `.fpp` can be included or whatnot